Compilation issues when test feature enabled

```
error[E0308]: mismatched types
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:106:12
    |
106 |     if let Some(time) = config.time {
    |            ^^^^^^^^^^   ----------- this match expression has type `log::LevelFilter`
    |            |
    |            expected enum `log::LevelFilter`, found enum `std::option::Option`
    |
    = note: expected type `log::LevelFilter`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:112:12
    |
112 |     if let Some(level) = config.level {
    |            ^^^^^^^^^^^   ------------ this match expression has type `log::LevelFilter`
    |            |
    |            expected enum `log::LevelFilter`, found enum `std::option::Option`
    |
    = note: expected type `log::LevelFilter`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:118:12
    |
118 |     if let Some(thread) = config.thread {
    |            ^^^^^^^^^^^^   ------------- this match expression has type `log::LevelFilter`
    |            |
    |            expected enum `log::LevelFilter`, found enum `std::option::Option`
    |
    = note: expected type `log::LevelFilter`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:124:12
    |
124 |     if let Some(target) = config.target {
    |            ^^^^^^^^^^^^   ------------- this match expression has type `log::LevelFilter`
    |            |
    |            expected enum `log::LevelFilter`, found enum `std::option::Option`
    |
    = note: expected type `log::LevelFilter`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:130:12
    |
130 |     if let Some(location) = config.location {
    |            ^^^^^^^^^^^^^^   --------------- this match expression has type `log::LevelFilter`
    |            |
    |            expected enum `log::LevelFilter`, found enum `std::option::Option`
    |
    = note: expected type `log::LevelFilter`
               found type `std::option::Option<_>`

error[E0599]: no method named `unwrap_or` found for type `std::borrow::Cow<'static, str>` in the current scope
   --> /home/jgoerzen/.cargo/registry/src/github.com-1ecc6299db9ec823/simplelog-0.7.1/src/loggers/testlog.rs:144:44
    |
144 |         cur_time.format(config.time_format.unwrap_or("%H:%M:%S"))
    |                                            ^^^^^^^^^

error: aborting due to 6 previous errors
```

Builds fine without test feature... but I need that feature

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,18 +13,36 @@ rust:
 - nightly
 
 matrix:
+  fast_finish: true
   allow_failures:
   - rust: nightly
 
+env:
+  matrix:
+    - FEATURES=""
+    - FEATURES="test"
+    - FEATURES="term"
+    - FEATURES="default"
+    - FEATURES="all"
+
 before_cache: |
-  if [[ "$TRAVIS_RUST_VERSION" == nightly ]]; then
-    RUSTFLAGS="--cfg procmacro2_semver_exempt" cargo install cargo-tarpaulin
+  if [[ "$TRAVIS_RUST_VERSION" == stable ]] && [[ "$FEATURES" == all ]]; then
+    cargo install cargo-tarpaulin
   fi
 
 script: |
-  cargo test --all --all-features
+  case $FEATURES in
+    "all")
+        cargo test --all --all-features
+    ;;
+    "default")
+        cargo test --all
+    ;;
+    *)
+      cargo test --all --no-default-features --features "$FEATURES"
+    esac
 
 after_success: |
-  if [[ "$TRAVIS_RUST_VERSION" == nightly ]]; then
+  if [[ "$TRAVIS_RUST_VERSION" == stable ]] && [[ "$FEATURES" == all ]]; then
     cargo tarpaulin --all-features --ciserver travis-ci --coveralls $TRAVIS_JOB_ID
   fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,5 +26,5 @@ script: |
 
 after_success: |
   if [[ "$TRAVIS_RUST_VERSION" == nightly ]]; then
-    cargo tarpaulin --ciserver travis-ci --coveralls $TRAVIS_JOB_ID
+    cargo tarpaulin --all-features --ciserver travis-ci --coveralls $TRAVIS_JOB_ID
   fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -22,7 +22,7 @@ before_cache: |
   fi
 
 script: |
-  cargo test --all
+  cargo test --all --all-features
 
 after_success: |
   if [[ "$TRAVIS_RUST_VERSION" == nightly ]]; then

--- a/examples/usage.rs
+++ b/examples/usage.rs
@@ -5,7 +5,10 @@ use std::fs::File;
 
 fn main() {
     CombinedLogger::init(vec![
+        #[cfg(feature = "term")]
         TermLogger::new(LevelFilter::Warn, Config::default(), TerminalMode::Mixed).unwrap(),
+        #[cfg(not(feature = "term"))]
+        SimpleLogger::new(LevelFilter::Warn, Config::default()),
         WriteLogger::new(
             LevelFilter::Info,
             Config::default(),

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -121,6 +121,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("error_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                vec.push(TestLogger::new(LevelFilter::Error, conf.clone()));
 
                 //warn
                 vec.push(
@@ -135,6 +136,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("warn_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                vec.push(TestLogger::new(LevelFilter::Warn, conf.clone()));
 
                 //info
                 vec.push(
@@ -149,6 +151,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("info_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                vec.push(TestLogger::new(LevelFilter::Info, conf.clone()));
 
                 //debug
                 vec.push(
@@ -163,6 +166,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("debug_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                vec.push(TestLogger::new(LevelFilter::Debug, conf.clone()));
 
                 //trace
                 vec.push(
@@ -177,6 +181,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("trace_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                vec.push(TestLogger::new(LevelFilter::Trace, conf.clone()));
             }
 
             vec

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -112,6 +112,7 @@ mod tests {
                 vec.push(
                     SimpleLogger::new(LevelFilter::Error, conf.clone()) as Box<dyn SharedLogger>
                 );
+                #[cfg(feature = "term")]
                 vec.push(
                     TermLogger::new(LevelFilter::Error, conf.clone(), TerminalMode::Mixed).unwrap()
                         as Box<dyn SharedLogger>,
@@ -121,12 +122,14 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("error_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                #[cfg(feature = "test")]
                 vec.push(TestLogger::new(LevelFilter::Error, conf.clone()));
 
                 //warn
                 vec.push(
                     SimpleLogger::new(LevelFilter::Warn, conf.clone()) as Box<dyn SharedLogger>
                 );
+                #[cfg(feature = "term")]
                 vec.push(
                     TermLogger::new(LevelFilter::Warn, conf.clone(), TerminalMode::Mixed).unwrap()
                         as Box<dyn SharedLogger>,
@@ -136,12 +139,14 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("warn_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                #[cfg(feature = "test")]
                 vec.push(TestLogger::new(LevelFilter::Warn, conf.clone()));
 
                 //info
                 vec.push(
                     SimpleLogger::new(LevelFilter::Info, conf.clone()) as Box<dyn SharedLogger>
                 );
+                #[cfg(feature = "term")]
                 vec.push(
                     TermLogger::new(LevelFilter::Info, conf.clone(), TerminalMode::Mixed).unwrap()
                         as Box<dyn SharedLogger>,
@@ -151,12 +156,14 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("info_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                #[cfg(feature = "test")]
                 vec.push(TestLogger::new(LevelFilter::Info, conf.clone()));
 
                 //debug
                 vec.push(
                     SimpleLogger::new(LevelFilter::Debug, conf.clone()) as Box<dyn SharedLogger>
                 );
+                #[cfg(feature = "term")]
                 vec.push(
                     TermLogger::new(LevelFilter::Debug, conf.clone(), TerminalMode::Mixed).unwrap()
                         as Box<dyn SharedLogger>,
@@ -166,12 +173,14 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("debug_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                #[cfg(feature = "test")]
                 vec.push(TestLogger::new(LevelFilter::Debug, conf.clone()));
 
                 //trace
                 vec.push(
                     SimpleLogger::new(LevelFilter::Trace, conf.clone()) as Box<dyn SharedLogger>
                 );
+                #[cfg(feature = "term")]
                 vec.push(
                     TermLogger::new(LevelFilter::Trace, conf.clone(), TerminalMode::Mixed).unwrap()
                         as Box<dyn SharedLogger>,
@@ -181,6 +190,7 @@ mod tests {
                     conf.clone(),
                     File::create(&format!("trace_{}.log", i)).unwrap(),
                 ) as Box<dyn SharedLogger>);
+                #[cfg(feature = "test")]
                 vec.push(TestLogger::new(LevelFilter::Trace, conf.clone()));
             }
 

--- a/src/loggers/comblog.rs
+++ b/src/loggers/comblog.rs
@@ -36,6 +36,7 @@ impl CombinedLogger {
     /// # fn main() {
     /// let _ = CombinedLogger::init(
     ///             vec![
+    /// #               #[cfg(feature = "term")]
     ///                 TermLogger::new(LevelFilter::Info, Config::default(), TerminalMode::Mixed).unwrap(),
     ///                 WriteLogger::new(LevelFilter::Info, Config::default(), File::create("my_rust_bin.log").unwrap())
     ///             ]
@@ -66,6 +67,7 @@ impl CombinedLogger {
     /// # fn main() {
     /// let combined_logger = CombinedLogger::new(
     ///             vec![
+    /// #               #[cfg(feature = "term")]
     ///                 TermLogger::new(LevelFilter::Debug, Config::default(), TerminalMode::Mixed).unwrap(),
     ///                 WriteLogger::new(LevelFilter::Info, Config::default(), File::create("my_rust_bin.log").unwrap())
     ///             ]

--- a/src/loggers/testlog.rs
+++ b/src/loggers/testlog.rs
@@ -8,7 +8,7 @@
 //! Module providing the TestLogger Implementation
 
 use super::logging::should_skip;
-use crate::{Config, SharedLogger};
+use crate::{Config, SharedLogger, LevelPadding};
 use log::{set_boxed_logger, set_max_level, LevelFilter, Log, Metadata, Record, SetLoggerError};
 
 use std::thread;
@@ -103,34 +103,24 @@ pub fn log(config: &Config, record: &Record<'_>) {
         return;
     }
 
-    if let Some(time) = config.time {
-        if time <= record.level() {
-            write_time(config);
-        }
+    if config.time <= record.level() && config.time != LevelFilter::Off {
+        write_time(config);
     }
 
-    if let Some(level) = config.level {
-        if level <= record.level() {
-            write_level(record);
-        }
+    if config.level <= record.level() && config.level != LevelFilter::Off {
+        write_level(record, config);
     }
 
-    if let Some(thread) = config.thread {
-        if thread < record.level() {
-            write_thread_id();
-        }
+    if config.thread < record.level() && config.thread != LevelFilter::Off {
+        write_thread_id();
     }
 
-    if let Some(target) = config.target {
-        if target <= record.level() {
-            write_target(record);
-        }
+    if config.target <= record.level() && config.target != LevelFilter::Off {
+        write_target(record);
     }
 
-    if let Some(location) = config.location {
-        if location <= record.level() {
-            write_location(record);
-        }
+    if config.location <= record.level() && config.location != LevelFilter::Off {
+        write_location(record);
     }
 
     write_args(record);
@@ -138,16 +128,24 @@ pub fn log(config: &Config, record: &Record<'_>) {
 
 #[inline(always)]
 pub fn write_time(config: &Config) {
-    let cur_time = chrono::Utc::now();
+    let cur_time = if config.time_local {
+        chrono::Local::now().naive_local() + config.time_offset
+    } else {
+        chrono::Utc::now().naive_utc() + config.time_offset
+    };
     print!(
         "{} ",
-        cur_time.format(config.time_format.unwrap_or("%H:%M:%S"))
+        cur_time.format(&*config.time_format)
     );
 }
 
 #[inline(always)]
-pub fn write_level(record: &Record<'_>) {
-    print!("[{}] ", record.level());
+pub fn write_level(record: &Record<'_>, config: &Config) {
+    match config.level_padding {
+        LevelPadding::Left => print!("[{: >5}] ", record.level()),
+        LevelPadding::Right => print!("[{: <5}] ", record.level()),
+        LevelPadding::Off => print!("[{}] ", record.level()),
+    };
 }
 
 #[inline(always)]

